dask/array/_shuffle.py

```diff
@@ -10,15 +10,20 @@
 import numpy as np
 
 from dask import config
-from dask._task_spec import DataNode, List, Task, TaskRef
+from dask._task_spec import DataNode, GraphNode, List, Task, TaskRef
 from dask.array.chunk import getitem
 from dask.array.core import Array, unknown_chunk_message
 from dask.array.dispatch import concatenate_lookup, take_lookup
 from dask.base import tokenize
 from dask.highlevelgraph import HighLevelGraph
 
 
-def shuffle(x, indexer: list[list[int]], axis: int, chunks: Literal["auto"] = "auto"):
+def shuffle(
+    x: Array,
+    indexer: list[list[int]],
+    axis: int,
+    chunks: Literal["auto"] | tuple[tuple[int, ...], ...] = "auto",
```

Comment:
shouldn't this be a `dict[int, tuple[tuple[int, ...], ...]]`? where `assert axis not in chunks`? For example, What if I was shuffling axis=1 of a 3D array, and wanted consistent chunks on the other axes.